server/runtime/src/main/resources/schema/infinispan-server-12.0.xsd

```diff
@@ -818,6 +825,108 @@
+      </xs:sequence>
+   </xs:complexType>
+
+   <xs:complexType name="credential-reference">
+      <xs:annotation>
+         <xs:documentation>
+            Credential reference to be used by the configuration.
+         </xs:documentation>
+      </xs:annotation>
+      <xs:attribute name="store" type="xs:string">
+         <xs:annotation>
+            <xs:documentation>
+               Credential store name used to fetch credential with given 'alias' from.
+               Credential store name has to be defined elsewhere.
+            </xs:documentation>
+         </xs:annotation>
+      </xs:attribute>
+      <xs:attribute name="alias" type="xs:string">
+         <xs:annotation>
+            <xs:documentation>
+               Alias of credential in the credential store.
+            </xs:documentation>
+         </xs:annotation>
+      </xs:attribute>
+   </xs:complexType>
+
+   <xs:complexType name="clear-text-credential">
+      <xs:annotation>
+         <xs:documentation>
+            A clear-text credential.
+         </xs:documentation>
+      </xs:annotation>
+      <xs:attribute name="clear-text" type="xs:string" use="required">
+         <xs:annotation>
+            <xs:documentation>
+               The clear-text password.
+            </xs:documentation>
+         </xs:annotation>
+      </xs:attribute>
+   </xs:complexType>
+
+   <xs:complexType name="credential-store">
+      <xs:annotation>
+         <xs:documentation>
+            An individual credential store definition.
+         </xs:documentation>
+      </xs:annotation>
+      <xs:choice>
+         <xs:element name="clear-text-credential" type="tns:clear-text-credential">
+            <xs:annotation>
+               <xs:documentation>
+                  Credential to be used by as protection parameter for the Credential Store.
+               </xs:documentation>
+            </xs:annotation>
+         </xs:element>
+         <xs:element name="credential-reference" type="tns:credential-reference">
+            <xs:annotation>
+               <xs:documentation>
+                  Credential to be used by as protection parameter for the Credential Store.
+               </xs:documentation>
+            </xs:annotation>
+         </xs:element>
+      </xs:choice>
+      <xs:attribute name="name" type="xs:string" use="required">
+         <xs:annotation>
+            <xs:documentation>
+               The name of this credential store.
+            </xs:documentation>
+         </xs:annotation>
+      </xs:attribute>
+      <xs:attribute name="relative-to" type="xs:string">
+         <xs:annotation>
+            <xs:documentation>
+               A reference to a previously defined path that the file name is
+               relative to.
+            </xs:documentation>
+         </xs:annotation>
+      </xs:attribute>
+      <xs:attribute name="path" type="xs:string" use="required">
```

Comment:
I guess it isn't a path but the file name... what about replacing with `file`, `store-file`, or `credentials-file`?

Reply:
it's a relative path.

Reply:
we have already a `relative-to` attribute. Having a `path` that is relative to `relative-to` (right?) seems a bit too much IMO.

Reply:
Every single other reference to a file in our config uses the `path`/`relative-to` pair. Remember that `relative-to` is not a path directly but the name of a property. This mimics the behaviour of WildFly where, for each file in the config, the `relative-to` attribute is set to the appropriate default (e.g. `infinispan.server.config.path` or `infinispan.server.data.path`) without having to include it in the `path` attribute directly (e.g. `${infinispan.server.config.path}/credentials.pfx`).
We could remove all traces of `relative-to`: if a `path` is relative, it will be relative to the default path for that element.

Reply:
ok. fair enough. Let's keep both.